Source/AppAuthCore/OIDAuthorizationService.m

```diff
@@ -107,15 +107,17 @@ - (void)cancelWithCompletion:(nullable void (^)(void))completion {
 + (BOOL)URL:(NSURL *)URL matchesRedirectionURL:(NSURL *)redirectionURL {
   NSURL *standardizedURL = [URL standardizedURL];
   NSURL *standardizedRedirectURL = [redirectionURL standardizedURL];
+  NSString *trimmedStdUrlPath = [standardizedURL.path stringByTrimmingCharactersInSet:[NSCharacterSet characterSetWithCharactersInString:@"/"]];
+  NSString *stdRedirectUrlPath = [standardizedRedirectURL.path stringByTrimmingCharactersInSet:[NSCharacterSet characterSetWithCharactersInString:@"/"]];
```

Comment:
Rather than trimming forward slashes from both ends of the paths and comparing, I think we should look at a narrower approach that modifies the existing check to consider a path value of "/" as equal to an empty path of "".  One way to achieve this would be to normalize "/" to the empty path "".  This would look something like:

```
// For this URL comparison, normalize path values of "/" to "".
NSString *normalizedPath =
    [standardizedURL.path isEqualToString:@"/"] ? @"" : standardizedURL.path;
NSString *normalizedRedirectPath =
    [standardizedRedirectURL.path isEqualToString:@"/"] ? @"" : standardizedRedirectURL.path;

...

return
...
    && OIDIsEqualIncludingNil(normalizedPath, normalizedRedirectPath);
```

Reply:
> Rather than trimming forward slashes from both ends of the paths and comparing, I think we should look at a narrower approach that modifies the existing check to consider a path value of "/" as equal to an empty path of "". One way to achieve this would be to normalize "/" to the empty path "". This would look something like:
> 
> ```
> // For this URL comparison, normalize path values of "/" to "".
> NSString *normalizedPath =
>     [standardizedURL.path isEqualToString:@"/"] ? @"" : standardizedURL.path;
> NSString *normalizedRedirectPath =
>     [standardizedRedirectURL.path isEqualToString:@"/"] ? @"" : standardizedRedirectURL.path;
> 
> ...
> 
> return
> ...
>     && OIDIsEqualIncludingNil(normalizedPath, normalizedRedirectPath);
> ```

Thank you for suggestion @petea . Not sure that this approach will help in cases, when a path is not just a trailing slash.
So, if a path is 'foo' and the server side will add a trailing slash, we'll compare 'foo' and 'foo/' and get the same issue again.
Am I missed anything?

For my particular case you suggestion will perfectly work, because my app doesn't have any path in callback url. But I'm still not sure in general case.

Reply:
The empty path case ("" or "/") is the only one in which the two URLs are arguably equivalent.  Adding a trailing slash to a non-empty path component ("/path" -> "/path/") changes the resource to which the URL points, so I don't think we should equate the two.  Luckily, in practice, we seem to be dealing only with the empty path case here.

Reply:
Ok. Your suggestion is already applied in this PR. My senior colleagues absolutely agree with you as well.